### DISCUSSION TITLE
SCRIPTING: Values refreshed before saving. How to fix

### DISCUSSION THREAD

**Discussion by UBWH** [arendst/Tasmota#20656] [General]
I have a script with a sub-webpage as below
```
>w3 Alarms
%/tasmota.css
%/alarms.htm
```

The alarms.htm page is:
```
...
nm(0 30 1 ablt "" 70 1)
...
```

This generates this html:
```
<input  min='0.0' max='30.0' step='1.0' value='30.0' type='number' style='width:70px' onfocusin='pr(0)' onfocusout='pr(1)' onchange='siva(value,"ablt")'>
```

Problem is:  the _ablt_ value is displayed correctly, but if changed by the user it quickly reverts back to the old value as the page is refreshing.

@gemu: Is there a way to stop the value on the screen changing?

**Comment by gemu2015** [comment]
i tried to reproduce , but i couldn't .
did you forget to define the numeric variable ablt ?

**Comment by UBWH** [comment]
I tried adding the "$" but does not help.

With "$": `nm(0 30 1 ablt "" 70 1)` in the HTML
No "$": `<input  min='0.0' max='30.0' step='1.0' value='30.0' type='number' style='width:70px' onfocusin='pr(0)' onfocusout='pr(1)' onchange='siva(value,"ablt")'>` in the HTML

Yes: ablt is defined in the >D section

The problem seems to be including the alarms.htm file is  not processing the `$` (but it IS processing the `nm()`

Here are some code snippets

SCRIPT
```
>D 48
;Alarms
ast=0
ablt=0

>w3 Alarms
%/tasmota.css
%/alarms.htm
```

UFS alarms.htm
```
<div style='text-align:left;display:inline-block;color:#eaeaea;min-width:340px;'>
<div style='text-align:left;color:#eaeaea;'>
<h1>Alarms</h1>
<p></p>
<fieldset>
<legend><b> Triggers </b></legend>
<table>
<tr><th>Battery Voltage<td>
$nm(0 30 1 ablt "" 70 1)
</table>
<hr>
<div>
<button type='submit' style="width:100%%" onclick='seva(1,"ast");history.back()'>Save</button>
</div>
</fieldset>
</div>

<p></p>
<form style="display: block;" action='.' method='get'><button>Main Menu</button></form>
</div>
```

**Reply by gemu2015** [reply]
the $ is not the right choice here it is needed for google chart macros

as said, i can not reproduce with 3 different browsers, your subfile works as expected

**Reply by gemu2015** [reply]
as you can see in the expanded macro it disables auto refresh when it has the focus.
may be your css file does interfere here. without css file it works

**Comment by UBWH** [comment]
Found the problem.

Sorry to waste your time.
